org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalReplacementProvider.java

```diff
@@ -214,10 +215,15 @@ private void appendAnonymousClass(StringBuilder completionBuffer, CompletionProp
 			if (!(element instanceof IType)) {
 				return;
 			}
-			IType type = (IType) element;
 			int offset = proposal.getReplaceStart();
-			AnonymousTypeCompletionProposal overrider = new AnonymousTypeCompletionProposal(compilationUnit, offset, type, String.valueOf(proposal.getDeclarationSignature()), client.isCompletionSnippetsSupported());
-			String replacement = overrider.updateReplacementString(document, offset, importRewrite);
+
+			if (this.anonymousTypeNewBody == null) {
+				// calculate and format an empty new body
+				AnonymousTypeCompletionProposal overrider = new AnonymousTypeCompletionProposal(compilationUnit, offset, client.isCompletionSnippetsSupported());
+				this.anonymousTypeNewBody = overrider.updateReplacementString(document, offset);
```

Comment:
These two lines are used to generate the code snippet
```
"(){
	${0}
};"
```

Since it's same for all anonymous type completion proposals, reusing it works well.